assets/src/hooks/useRoutePatterns.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react"
+import { fetchRoutePatterns } from "../api"
+import { RouteId, RoutePattern } from "../schedule"
+
+const useRoutePatterns = (routeId: RouteId | null): RoutePattern[] | null => {
+  const [routePatterns, setRoutePatterns] = useState<RoutePattern[] | null>(
+    null
+  )
+  useEffect(() => {
+    if (routeId === null) {
+      setRoutePatterns([])
```

Comment:
**question:** should this be `null` instead? Also, what about the race condition where you call `useRoutePatterns("some route")` and kick off a fetch, then immediately call `useRoutePatterns(null)`, but then the promise from the first call resolves and sets the route patterns to the value returned for `"some route"`?

Reply:
Thanks for catching this race condition! On whether it should be null in this case, I've been thinking about null as representing data that is loading, but in this instance, there is no data to load. 

Both of these questions make me think about the `useFetch` library @firestack explored adding in a previous PR - looking at the docs it seems to handle the race conditions and clearer loading state. I may look into using that if it seems appropriate / reasonably low effort

Reply:
Opted for a lower-effort approach in [5add81d](https://github.com/mbta/skate/pull/1904/commits/5add81d752c260d82b89bb0632e3905ef40161b3)